File: tags/emberplussdk-1-4-0/EmberLib.net/EmberPlusRouter/Model/IntegerParameter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace EmberPlusRouter.Model
{
   class IntegerParameter : Parameter<long>
   {
      public IntegerParameter(int number, Element parent, string identifier, Dispatcher dispatcher, int min, int max, bool isWriteable)
      : base(number, parent, identifier, dispatcher, isWriteable)
      {
         Minimum = min;
         Maximum = max;
      }

      public long Minimum { get; private set; }
      public long Maximum { get; private set; }

      public override TResult Accept<TState, TResult>(IElementVisitor<TState, TResult> visitor, TState state)
      {
         return visitor.Visit(this, state);
      }
   }
}

File: branches/emberplussdk-1-4-branch/tinyember/TinyEmberPlusRouter/TinyEmberPlusRouter/model/IntegerParameter.cpp
#include "ElementVisitor.h"
#include "IntegerParameter.h"

namespace model
{
   IntegerParameter::IntegerParameter(int number, Element* parent, std::string identifier, NotificationSink* notificationSink, int minimum, int maximum)
      : Parameter(number, parent, identifier, notificationSink)
      , m_minimum(minimum)
      , m_maximum(maximum)
   {}

   void IntegerParameter::accept(ElementVisitor* visitor)
   {
      visitor->visit(this);
   }

   void IntegerParameter::onValueChanged()
   {
      auto notificationSink = this->notificationSink();

      if(notificationSink != nullptr)
         notificationSink->notifyParameterValueChanged(path(), value());
   }
}
